src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -236,6 +241,19 @@ export const useSearch = (services: DiscoverViewServices) => {
           elapsedMs,
         },
       });
+
+      // Add tracking for datasource type and query if the query is not same as last query
+      const query = searchSource.getField('query');
+      const queryStringHistory = data.query.queryString.getQueryHistory();
+      if (
```

Comment:
@kavilla . That actually seems more neat. Will move the tracking to over there.

Reply:
apologies i wasn't too clear with my suggestion. 🤕 

what i meant that here you are access the data plugin to get the information to track it and which point you are interacting with the value after you get the response.

but instead of accessing the data plugin after you get a response, you move this logic within the data plugin where it has context of the previous query and the current query and the link above was more so to indicate how it has access to the previous query. so we could add a function in the data plugin that reports a ui metric if there was a new query. the ui service could even just subscribe to the query service or maybe the query service itself reports the metric since it already has access to the querystring history